### ISSUE TITLE
add missing file extension to jpegs in article xml

### ISSUE THREAD

**Issue by GiancarloFusiello** [libero/libero#200]
eLife article 40092 currently raises an error when trying to be read by the scholarly articles service due to missing file extension of jpeg images, causing the uri to the file to be incorrect.

**Task**
- create a task to add `.jpg` extension where missing in the article xml

**Comment by Melissa37** [comment]
Sorry to jump in, but is it a given that Libero will convert any image provided into a .jpeg if not already?

It's a bit dodgy to start messing with publisher XML.

Would it be better to have a requirement that the XML contains the file extension type in the XML?

But what if it's a .tif or something else because that is what is used in production? Does Libero require .jpegs to be submitted and the file extensions to be in the XML?

I don't think this is a decision to just be made without thinking through all the potential issues. I know it gets us to a site quicker, but it might require more work later on...

**Comment by thewilkybarkid** [comment]
`xlink:href` _requires_ the correct path and filename (including the extension), so eLife's usage without is wrong (it's pointing to a file that doesn't exist). The requirement has to be [that the file exists](https://www.w3.org/TR/xlink/#link-locators) (@FAtherden-eLife mentioned somewhere about using Schematron to check).

(Refs https://github.com/libero/jats-ingester/pull/30#issuecomment-507625549 and https://github.com/libero/jats-ingester/pull/30#issuecomment-507626965.)

My concern here is spending time writing code to get round data problems, when just tweaking the file by hand is definitely quicker (since the volume for the demo is tiny). If there is a reason to make changes reproducible, @FAtherden-eLife could probably quickly produce some XSLT since these are production issues and should be fixed there too. /cc @BlueReZZ

**Comment by Melissa37** [comment]
Sorry, my point is that the system should not be correcting XML. The XML should be right when submitted, so we should not be building Libero Publisher to do things like this, we should be ensuring Publishers give us the correct XML and that should be an input requirement.

We're not building this for eLife.

Schematron is built by publishers in their publisher workflows and we cannot say everyone will use the same and have the same processes we have.

In order for Libero to be extensible, it should have that as an input requirement rather than correcting someone's XML and creating a different version of XML, which is not a good thing (in my opinion).

If eLife's XML is the only wrong XML, we should correct it at source. If other publishers are incorrect too, we need to think about why that is and talk to people rather than make an assumption to create a tool to fix it....

**Comment by Melissa37** [comment]
Ah @thewilkybarkid just re-read your comment - we are on the same page :-)

**Comment by GiancarloFusiello** [comment]
@BlueReZZ this requires your attention. Please see above. Blocking until a decision is made to whether to add this feature.

https://github.com/libero/jats-ingester/pull/30 is ready to be merged when unblocked if necessary.

**Comment by BlueReZZ** [comment]
Since this is done in https://github.com/libero/jats-ingester/pull/30 which is a single, separate job in the ingester I think it's prudent to merge that to get around the problem in the knowledge that we will remove it once the XML is fixed.

We should record the need to remove it in a separate issue with the precondition that the XML is fixed (it would also be good if we could link to an issue in the appropriate place that covers to work to fix the XML)
